docs/changes/coordinates/16246.feature.rst

```diff
@@ -0,0 +1,9 @@
+The ``SkyCoord`` and ``BaseCoordinateFrame`` ``separation()`` methods now
+accept an optional ``frame_mismatch`` argument.
```

Comment:
Together with my previous suggestion to make the argument keyword-only from the get go.
```suggestion
accept an optional ``frame_mismatch`` keyword argument.
```